models.py

```diff
@@ -35,62 +27,28 @@ def _create_callbacks(self, **kwargs):
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_convo, l2_reg_dense, use_prelu, use_batch_norm)
-
-
-class WordLSTM(CharacterClassifier):
-    def _build_optimizer(self, **kwargs):
-        return Adam(kwargs['lr'])
-
-    def _build_model(self, embedding_size,
-                     maxlen, max_features,
-                     unit_nr, repeat_block, dropout_lstm,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_dense, use_prelu, use_batch_norm, global_pooling):
-        return lstm(None, embedding_size,
-                    maxlen, max_features,
-                    unit_nr, repeat_block, dropout_lstm,
-                    dense_size, repeat_dense, dropout_dense,
-                    l2_reg_dense, use_prelu, use_batch_norm, False, global_pooling)
-
-
-class WordDPCNN(CharacterClassifier):
-    def _build_optimizer(self, **kwargs):
-        return SGD(**kwargs)
-
-    def _build_model(self, embedding_size,
-                     maxlen, max_features,
-                     filter_nr, kernel_size, repeat_block, dropout_convo,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_convo, l2_reg_dense, use_prelu, trainable_embedding, use_batch_norm):
-        """
-        Implementation of http://ai.tencent.com/ailab/media/publications/ACL3-Brady.pdf
-        """
-        return dpcnn(None, embedding_size,
-                     maxlen, max_features,
-                     filter_nr, kernel_size, repeat_block, dropout_convo,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_convo, l2_reg_dense, use_prelu, trainable_embedding, use_batch_norm)
-
-
-class GloveBasic(CharacterClassifier):
+class CharVDCNN(BasicClassifier):
+    def _build_model(self, embedding_size, maxlen, max_features,
+                     filter_nr, kernel_size, repeat_block, dense_size, repeat_dense,
+                     max_pooling, mean_pooling, weighted_average_attention, concat_mode,
+                     dropout_embedding, conv_dropout, dense_dropout, dropout_mode,
+                     conv_kernel_reg_l2, conv_bias_reg_l2,
+                     dense_kernel_reg_l2, dense_bias_reg_l2,
+                     use_prelu, use_batch_norm, batch_norm_first):
+        return vdcnn(embedding_size, maxlen, max_features,
+                     filter_nr, kernel_size, repeat_block, dense_size, repeat_dense,
+                     max_pooling, mean_pooling, weighted_average_attention, concat_mode,
+                     dropout_embedding, conv_dropout, dense_dropout, dropout_mode,
+                     conv_kernel_reg_l2, conv_bias_reg_l2,
+                     dense_kernel_reg_l2, dense_bias_reg_l2,
+                     use_prelu, use_batch_norm, batch_norm_first)
+
+
+class PretrainedEmbeddingModel(BasicClassifier):
     def fit(self, embedding_matrix, X, y, validation_data):
```

Comment:
`embedding_matrix` should be last arg, then `fit(self, X, y, validation_data)` in the `ClassifierXY` should also, accept additional args. This is because, we want to override `ClassifierXY`, but keep its definition.

Reply:
@jakubczakon in the
```
class PretrainedEmbeddingModel(BasicClassifier):
    def fit(self, embedding_matrix, X, y, validation_data):
```

Change order of args to this: `fit(self, X, y, validation_data, embedding_matrix)`. Also in the `ClassifierXY add **kwargs to the definition, so that it can accept additional params when needed. It is more coherent solution.